src/Directory.Build.targets

```diff
@@ -10,5 +14,119 @@
     <EmbeddedResource Update="**\*.resx" GenerateSource="true" />
   </ItemGroup>
 
-  <Import Project="$(RepoToolsetDir)Imports.targets" Condition="Exists('$(RepoToolsetDir)Imports.targets')" />  
+  <ItemGroup>
```

Comment:
This adds significant amount of infrastructure code, which smells like something that should probably belong to RepoToolset, instead of individual repos? Are we planning to eventually move this code over?
@tmat is probably the correct person to review and comment on these changes.

Reply:
> I should read the PR description properly :-). Are we planning to merge in this change and revert once the functionality is in RepoToolset or wait for a RepoToolset package with this functionality before cleaning and merging this PR?

I would like to get it merged so we can get a better understanding of how this configuration of codecov.io works with our development process. Any tweaks we make can be included in dotnet/roslyn-tools#170 before it gets merged.